src/libs/antares/study/parameters.cpp

```diff
@@ -1020,24 +925,57 @@ bool Parameters::loadFromINI(const IniFile& ini, uint version, const StudyLoadOp
     reset();
     // A temporary buffer, used for the values in lowercase
     String value;
-
+    String sectionName;
+    typedef bool (*Callback)(Parameters&, const String&, const String&, const String&, uint);
+
+    static const std::map<String,Callback> sectionAssociatedToKeysProcess = {
+            {"general",&SGDIntLoadFamily_General},
+            {"input",&SGDIntLoadFamily_Input},
+            {"output",&SGDIntLoadFamily_Output},
+            {"optimization",&SGDIntLoadFamily_Optimization},
+            {"other preferences",&SGDIntLoadFamily_OtherPreferences},
+            {"advanced parameters",&SGDIntLoadFamily_AdvancedParameters},
+            {"playlist",&SGDIntLoadFamily_Playlist},
+            {"variables selection",&SGDIntLoadFamily_VariablesSelection},
+            {"seeds - mersenne twister",&SGDIntLoadFamily_SeedsMersenneTwister}};
+
+    Callback handleAllKeysInSection;
     // Foreach section on the ini file...
     for (auto* section = ini.firstSection; section; section = section->next)
     {
+        sectionName = section->name;
+        sectionName.toLower();
+        try
+        {
+            handleAllKeysInSection = sectionAssociatedToKeysProcess.at(sectionName);
+        } catch (const std::out_of_range&)
+        {
+            // Continue on error
+            logs.warning() << ini.filename() << ": '" << section->name << "': Unknown section name";
+            continue;
+        }
+
         // Foreach properties in the section
         for (const IniFile::Property* p = section->firstProperty; p; p = p->next)
         {
             if (p->key.empty())
                 continue;
+            if (!firstKeyLetterIsValid(p->key))
+                continue;
             // We convert the key and the value into the lower case format
             value = p->value;
             value.toLower();
 
             // Deal with the current property
-            if (not ParametersInternalLoadProperty(*this, p->key, value, p->value, version))
+            // Do not forget the variable `key` and `value` are identical to
+            // `p->key` and `p->value` except they are already in the lower case format
+            if (not (handleAllKeysInSection)(*this, p->key, value, p->value, version))
```

Comment:
Style : parenthesis around function name are not needed.

Reply:
Ok : 0b051f3